CMakeLists.txt

```diff
@@ -125,11 +125,27 @@ endif(JSON_ENABLED)
 # get the current git revision, if git is installed. Avoid warnings if it isn't
 FIND_PACKAGE(Git)
 IF(Git_FOUND)
+	EXECUTE_PROCESS(COMMAND ${GIT_EXECUTABLE} rev-parse --git-dir WORKING_DIRECTORY ${PROJECT_SOURCE_DIR} OUTPUT_VARIABLE git_dir OUTPUT_STRIP_TRAILING_WHITESPACE ERROR_QUIET)
+	IF(git_dir STREQUAL "${PROJECT_SOURCE_DIR}/.git")
+		SET(IS_GIT_REPO TRUE) 
+	ELSE()
+		SET(IS_GIT_REPO FALSE)
+	ENDIF()
+ELSE()
+	SET(IS_GIT_REPO FALSE)
+ENDIF(Git_FOUND)
+
+IF(IS_GIT_REPO)
 	EXECUTE_PROCESS(COMMAND ${GIT_EXECUTABLE} rev-parse --short HEAD WORKING_DIRECTORY ${PROJECT_SOURCE_DIR} OUTPUT_VARIABLE revision OUTPUT_STRIP_TRAILING_WHITESPACE)
 	EXECUTE_PROCESS(COMMAND ${GIT_EXECUTABLE} describe --tags --abbrev=0 HEAD WORKING_DIRECTORY ${PROJECT_SOURCE_DIR} OUTPUT_VARIABLE tag OUTPUT_STRIP_TRAILING_WHITESPACE)
 ELSE()
-	SET(revision GIT_NOT_FOUND)
-ENDIF(Git_FOUND)
+	get_filename_component(dir_name ${PROJECT_SOURCE_DIR} NAME)
+	string(REGEX MATCH "-([0-9]+\\.[0-9]+\\.[0-9]+.*)$" version_match ${dir_name})
+	if(version_match)
+		SET(tag ${CMAKE_MATCH_1})
+	endif()
```

Comment:
I've pushed a revision that sets revision to "none" for all non-git builds, and tag to "unknown" when nothing can be found.  It also uses the 'v' prefix for the non-git tag for consistency with git, and so that all versions will sort as being after "unknown".